discord/poll.py

```diff
@@ -0,0 +1,624 @@
+        return cls(text=data['text'])
+
+
+class PollAnswer:
+    """Represents a poll's answer.
+
+    .. container:: operations
+
+        .. describe:: str(x)
+
+            Returns this answer's text, if any.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The ID of this answer.
+    media: :class:`PollMedia`
+        A :class:`NamedTuple` containing the raw data of this answers media.
+    self_voted: :class:`bool`
+        Whether the current user has voted to this answer or not.
+    """
+
+    __slots__ = ('media', 'id', '_state', '_message', '_vote_count', 'self_voted', '_poll')
+
+    def __init__(
+        self,
+        *,
+        message: Optional[Message],
+        poll: Poll,
+        data: PollAnswerWithIDPayload,
+    ) -> None:
+        self.media: PollMedia = PollMedia.from_dict(data=data['poll_media'])
+        self.id: int = int(data['answer_id'])
+        self._message: Optional[Message] = message
+        self._state: Optional[ConnectionState] = message._state if message else None
+        self._vote_count: int = 0
+        self.self_voted: bool = False
+        self._poll: Poll = poll
+
+    def _handle_vote_event(self, added: bool) -> None:
+        if added:
+            self._vote_count += 1
+        else:
+            self._vote_count -= 1
+
+    def _update_with_results(self, payload: PollAnswerCountPayload) -> None:
+        self._vote_count = int(payload['count'])
+        self.self_voted = payload['me_voted']
+
+    def __str__(self) -> str:
+        return self.media.text
+
+    def __repr__(self) -> str:
+        return f'<PollAnswer id={self.id} media={self.media!r}>'
+
+    @classmethod
+    def from_params(
+        cls,
+        id: int,
+        text: str,
+        emoji: Optional[PollMediaEmoji] = None,
+        *,
+        poll: Poll,
+        message: Optional[Message],
+    ) -> Self:
+        poll_media: PollMediaPayload = {'text': text}
+        if emoji:
+            if isinstance(emoji, Emoji):
+                poll_media['emoji'] = {'name': emoji.name}  # type: ignore
+
+                if emoji.id:
+                    poll_media['emoji']['id'] = emoji.id  # type: ignore
+            elif isinstance(emoji, PartialEmoji):
+                poll_media['emoji'] = emoji.to_dict()
+            else:
+                poll_media['emoji'] = {'name': str(emoji)}  # type: ignore
+
+        payload: PollAnswerWithIDPayload = {'answer_id': id, 'poll_media': poll_media}
+
+        return cls(data=payload, message=message, poll=poll)
+
+    @property
+    def text(self) -> str:
+        """:class:`str`: Returns this answer display text."""
+        return self.media.text
+
+    @property
+    def emoji(self) -> Optional[Union[PartialEmoji, Emoji]]:
+        """Optional[:class:`PartialEmoji`]: Returns this answer display emoji, is any."""
+        if isinstance(self.media.emoji, str):
+            return PartialEmoji.from_str(self.media.emoji)
+        return self.media.emoji
```

Comment:
You're doing this here but this should have been done in the constructor for `PollMedia`.